pkg/common/utils.go

```diff
@@ -148,3 +148,36 @@ func CreateNodeID(project, zone, name string) string {
 func CreateZonalVolumeID(project, zone, name string) string {
 	return fmt.Sprintf(volIDZonalFmt, project, zone, name)
 }
+
+// ConvertLabelsStringToMap converts the labels from string to map
+// example: "key1=value1,key2=value2" gets converted into {"key1": "value1", "key2": "value2"}
+func ConvertLabelsStringToMap(labels string) (map[string]string, error) {
+	const labelsDelimiter = ","
+	const labelsKeyValueDelimiter = "="
+
+	m := make(map[string]string)
+	if labels == "" {
+		return m, nil
+	}
+
+	errIncorrectFormat := fmt.Errorf("labels %q are invalid, correct format: 'key1=value1,key2=value2'", labels)
+
+	keyValueStrings := strings.Split(labels, labelsDelimiter)
+	for _, keyValue := range keyValueStrings {
+		keyValue := strings.Split(keyValue, labelsKeyValueDelimiter)
+
```

Comment:
Do we want to validate the characters in the key/values? It would be nice to get immediate errors for this sort of thing rather than waiting for a possibly obscured error from a GCP response. While generally I don't think we want to duplicate GCP validation logic, doing some reasonable sanity checks might make sense.

eg from https://cloud.google.com/resource-manager/docs/creating-managing-labels:

Keys have a minimum length of 1 character and a maximum length of 63 characters, and cannot be empty. Values can be empty, and have a maximum length of 63 characters.
Keys and values can contain only lowercase letters, numeric characters, underscores, and dashes. All characters must use UTF-8 encoding, and international characters are allowed.
The key portion of a label must be unique. However, you can use the same key with multiple resources.
Keys must start with a lowercase letter or international character.